src/components/Counter.component.js

```diff
@@ -0,0 +1,39 @@
+import React, { Component } from 'react'
+// import PropTypes from 'prop-types'
+import './../styles/instructions.css'
+
+class Counter extends Component {
+  state = { count: this.props.initialCount }
+  handlePlusClick = () => {
+    this.setState(state => {
+      return { count: state.count + 1 }
+    })
+  }
+  handleReset = () => {
+    this.setState({ count: 0 })
+  }
+  handleMinusClick = () => {
+    this.setState(state => {
+      return { count: state.count - 1 }
+    })
+  }
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.initialCount !== this.state.count) {
```

Comment:
Nice! A lot of people miss this `if` check and it simply increases rendering time 👍 
However, I *think* you meant to do:
```javascript
if (nextProps.initialCount !== this.props.initialCount) {
```
 I think you want to compare it to current props, not the current state since you only want to execute this if the prop **changes** between renders